blocks/monitoring/evaluators.py

```diff
@@ -181,7 +182,13 @@ def initialize_aggregators(self):
         self.buffer_.initialize_aggregators()
 
     def process_batch(self, batch):
-        batch = dict_subset(batch, self.buffer_.input_names)
+        try:
+            batch = dict_subset(batch, self.buffer_.input_names)
+        except KeyError:
+            reraise_as(Exception(
```

Comment:
You should be able to just provide a string to `reraise_as`, there's no reason to change a different error type here, and theoretically it also means a `KeyError` can be caught again somewhere higher in the call stack (unlikely in this case, but anyway).